File: CS/Q182806/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace Q182806
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            DataSet dataSet11 = new DataSet();
            dataSet11.Tables.Add(GetCustomerDataTable());
            dataSet11.Tables.Add(GetPersonDataTable());
            DataColumn keyColumn = dataSet11.Tables["Customers"].Columns["ID"];
            DataColumn foreignKeyColumn = dataSet11.Tables["Persons"].Columns["ID"];
            dataSet11.Relations.Add("CustomersPersons", keyColumn, foreignKeyColumn);
            myGridControl1.DataSource = dataSet11.Tables["Customers"];
            myGridView1.KeyFieldName = "Items";
            myGridControl1.ForceInitialize();
        }
        DataTable GetCustomerDataTable()
        {
            DataTable table = new DataTable();
            table.TableName = "Customers";
            table.Columns.Add(new DataColumn("Items", typeof(string)));
            table.Columns.Add(new DataColumn("Money", typeof(double)));
            table.Columns.Add(new DataColumn("ID", typeof(int)));
            for (int i = 0; i < 10; i++)
                table.Rows.Add("Product " + i, 3000 + i * 298.55M, i);
            return table;
        }
        DataTable GetPersonDataTable()
        {
            DataTable table = new DataTable();
            table.TableName = "Persons";
            table.Columns.Add(new DataColumn("FirstName", typeof(string)));
            table.Columns.Add(new DataColumn("SecondName", typeof(string)));
            table.Columns.Add(new DataColumn("Age", typeof(int)));
            table.Columns.Add(new DataColumn("ID", typeof(int)));
            for (int i = 0; i < 50; i++)
            {
                string name = "Adam";
                string secondName = "Smith";
                if (i % 2 == 0)
                {
                    name = "Ben";
                    secondName = "Black";
                }
                table.Rows.Add(name, secondName, 20 + i / 2, i % 10);
            }
            return table;
        }
    }

}

File: VB/Q182806/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Namespace Q182806
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
			Dim dataSet11 As New DataSet()
			dataSet11.Tables.Add(GetCustomerDataTable())
			dataSet11.Tables.Add(GetPersonDataTable())
			Dim keyColumn As DataColumn = dataSet11.Tables("Customers").Columns("ID")
			Dim foreignKeyColumn As DataColumn = dataSet11.Tables("Persons").Columns("ID")
			dataSet11.Relations.Add("CustomersPersons", keyColumn, foreignKeyColumn)
			myGridControl1.DataSource = dataSet11.Tables("Customers")
			myGridView1.KeyFieldName = "Items"
			myGridControl1.ForceInitialize()
		End Sub
		Private Function GetCustomerDataTable() As DataTable
			Dim table As New DataTable()
			table.TableName = "Customers"
			table.Columns.Add(New DataColumn("Items", GetType(String)))
			table.Columns.Add(New DataColumn("Money", GetType(Double)))
			table.Columns.Add(New DataColumn("ID", GetType(Integer)))
			For i As Integer = 0 To 9
				table.Rows.Add("Product " & i, 3000 + i * 298.55D, i)
			Next i
			Return table
		End Function
		Private Function GetPersonDataTable() As DataTable
			Dim table As New DataTable()
			table.TableName = "Persons"
			table.Columns.Add(New DataColumn("FirstName", GetType(String)))
			table.Columns.Add(New DataColumn("SecondName", GetType(String)))
			table.Columns.Add(New DataColumn("Age", GetType(Integer)))
			table.Columns.Add(New DataColumn("ID", GetType(Integer)))
			For i As Integer = 0 To 49
				Dim name As String = "Adam"
				Dim secondName As String = "Smith"
				If i Mod 2 = 0 Then
					name = "Ben"
					secondName = "Black"
				End If
				table.Rows.Add(name, secondName, 20 + i \ 2, i Mod 10)
			Next i
			Return table
		End Function
	End Class

End Namespace